### PR TITLE
encrypt bottlerocket os volume

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Fixes https://github.com/weaveworks/eksctl/issues/2945

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [X] Manually tested
- [X] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [X] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by aclevername** [comment]
open to alternatives to `OSVolumeName` @Callisto13 @michaelbeaumont

**Review by Callisto13 — APPROVED** [review]
I think `OSVolumeName` is good, makes it clear what it is for which may save people the initial confusion we went through

File: pkg/cfn/builder/nodegroup.go [hunk 1]
```diff
@@ -148,6 +148,16 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 				Iops:       volumeIOPS,
 			},
 		}}
+
+		if n.spec.AdditionalEncryptedVolume != nil && *n.spec.AdditionalEncryptedVolume != "" {
```
Comment (michaelbeaumont) [inline pkg/cfn/builder/nodegroup.go:152]: we have `api.IsSetAndNonEmptyString`

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -1134,6 +1134,8 @@ type NodeGroupBase struct {
 	VolumeType *string `json:"volumeType,omitempty"`
 	// +optional
 	VolumeName *string `json:"volumeName,omitempty"`
+	// Some AMIs (bottlerocket) have a separate volume for the OS
+	AdditionalEncryptedVolume *string
```
Comment (michaelbeaumont) [inline pkg/apis/eksctl.io/v1alpha5/types.go:1138]: this can probably be just a `string` right? (or `[]string` :grin: )

Comment (aclevername) [reply pkg/apis/eksctl.io/v1alpha5/types.go:1138]: 🤣  I'll change it to a string

**Review by michaelbeaumont — APPROVED** [review]

**Comment by Callisto13** [comment]
@michaelbeaumont you wouldn't happen to be working while on holiday, would you? 👀  😏

**Comment by michaelbeaumont** [comment]
Just contributing to open source projects :grin:

**Comment by Callisto13** [comment]
Such a good citizen

**Pull request merged**